### ISSUE TITLE
Container.ForceCaptureInput can crash with Segmentation Fault while processing notifications under some unexpected circumstances

### ISSUE THREAD

**Issue by eugeneloza2factor** [castle-engine/castle-engine#588]
Not sure what exactly can go wrong here, because all of the below work separately without issues, it's in some obscure combination they crash. E.g. even simply removing absolutely unrelated `Update` and `Press` from the attached minimal example fixes the issue even though they are not related to the ForceCaptureInput in any way. But of what I could scrape:

1. Doesn't crash if `ForceCaptureInput` is a component in the current View design, only in a `TCastleDesign`
2. Doesn't crash if `ForceCaptureInput` component is owned by `FreeAtStop` (e.g. designed in the Editor), not by some component inside a TCastleDesign
3. Doesn't crash if `ForceCaptureInput` is freed manually, only through the helper.
4. Doesn't crash if no `Update` and no `Press` are present in the view, though they aren't related in any way.

Minimal reproduction:

[crash-forcecaptureinput.zip](https://github.com/castle-engine/castle-engine/files/14769046/crash-forcecaptureinput.zip)

1. Run the app
2. Click the Edit in the bottom-left to assign ForceCaptureInput to this Edit
3. Edit will get destroyed through TCastleUserInterfaceHelper.ClearAndFreeControls
4. Close the app -> crash

![2024-03-27 08_07_28-Greenshot](https://github.com/castle-engine/castle-engine/assets/73470521/ca8d0e0d-1b4b-4ff1-89c6-70712ed08557)

See also https://discord.com/channels/389676745957310465/389676745957310467/1222422043660652577

**Comment by michaliskambi** [comment]
Thank you! Fixed by using `TFreeNotificationObserver` internally, which we advise already throughout CGE ( https://castle-engine.io/modern_pascal#_free_notification_observer_castle_game_engine ).

I actually didn't debug it to the end... I initially assumed that the issue is that component assigned to `ForceCaptureInput` is also already owned by container. In which case our `TFreeNotificationObserver` works better (because standard `FreeNotification` will not "watch a component 2nd time", the 2nd instruction to watch is ignored). But after doing the change, I don't think this analysis applies to this case. Still, using `TFreeNotificationObserver` solved it, and we advise using `TFreeNotificationObserver` throughout CGE already. 

So I'm calling it a day for now -- I did something that was a good idea anyway (using `TFreeNotificationObserver`), and it helped. I may wake in the middle of some night and decide to debug it properly, but for now I can live with this resolution :)
